Asset Overview / Token Detail view redesign

nikoferro opened the pull request:
**Description**

Asset Overview / Asset Detail" page redesign involves some key changes: 

- Interactive chart will be added to the page which will display pricing information for the asset across different time periods. 
- Users will be able to review the prices through the price chart.

The redesign will also include the repositioning of the buy/send/receive/swap buttons.

**Screenshots/Recordings**

![New Project](https://user-images.githubusercontent.com/8658960/234217004-0efc16db-1e10-403e-adf7-cda6fb68d15d.png)

![Simulator Screen Recording - iPhone 11 - 2023-04-25 at 10 19 42](https://user-images.githubusercontent.com/8658960/234217486-eb7015df-ee9b-4e76-ab2e-09d1da764ff7.gif)

**Checklist**

* [ ] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

nikoferro committed:
feat: asset detail view

nikoferro committed:
Merge branch 'main' of github.com:nikoferro/metamask-mobile into swaps/token-detail

nikoferro committed:
Merge branch 'main' of github.com:nikoferro/metamask-mobile into swaps/token-detail

nikoferro committed:
feat: descriptions on detail view

nikoferro committed:
Merge branch 'main' of github.com:nikoferro/metamask-mobile into swaps/token-detail

nikoferro committed:
feat: price chart with tooltip

nikoferro committed:
feat: no data chart + icons fixed + tx list redesign

nikoferro committed:
Merge branch 'main' of github.com:nikoferro/metamask-mobile into swaps/token-detail

nikoferro committed:
chore: localizations for asset overview

nikoferro committed:
chore: removing old asset overview and renaming new one

nikoferro committed:
fix: chart price channel fixed

nikoferro committed:
fix: chart typing

nikoferro committed:
Merge branch 'main' of github.com:nikoferro/metamask-mobile into swaps/token-detail

nikoferro committed:
chore: analytics update on Asset view

nikoferro committed:
feat: added disclaimer for about text on asset overview

nikoferro committed:
chore: including disclaimer copy

nikoferro committed:
fix: about asset import fix

nikoferro committed:
chore: design feedback + new icons + test passing

nikoferro committed:
Merge branch 'main' of github.com:nikoferro/metamask-mobile into swaps/token-detail

nikoferro committed:
Merge branch 'main' of github.com:nikoferro/metamask-mobile into swaps/token-detail

nikoferro committed:
fix: moved to the new pricing api endpoint

nikoferro committed:
chore: revert changes related to icons

nikoferro committed:
chore: removing icon buttons

nikoferro committed:
feat: added shadow for navbar on scroll

nikoferro committed:
Merge branch 'main' of github.com:nikoferro/metamask-mobile into swaps/token-detail

nikoferro committed:
Merge branch 'main' of github.com:nikoferro/metamask-mobile into swaps/token-detail

nikoferro committed:
feat: remove price channel on chart and minor adjustments

nikoferro committed:
chore: updating asset overview disclaimer

nikoferro committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into swaps/token-details

nikoferro committed:
chore: audit fix

nikoferro committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into swaps/token-details

nikoferro committed:
fix: audit fix

nikoferro committed:
fix: adding d3-color to transformIgnorePatterns

nikoferro committed:
fix: tests for new function variant

nikoferro committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into swaps/token-details

nikoferro committed:
chore: missing snapshots

nikoferro committed:
fix: updated failing snapshots

nikoferro committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into swaps/token-details

nikoferro committed:
chore: make chart size smaller

nikoferro committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into swaps/token-details

nikoferro committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into swaps/token-details

davibroc commented:
I have tested this PR on browserstack on several iOS and Android devices, I also tested on my physical devices and reported the issues in the following doc which that were fixed and verified. @cortisiko unfortunately I don't have the rights to approve this PR or change the label to `QA Passe` but will ask @Andepande to help me with that and follow up with helpdesk
https://docs.google.com/document/d/1FOBfKeZx0NJc49Tuq6Jc1MO6zE4KIYwJ0npKooKRB50

sethkfman commented on locales/languages/en.json:
```diff
@@ -831,7 +831,30 @@
+    "unable_to_load_balance": "Unable to load your balance",
+    "about": "About",
+    "activity": "{{symbol}} activity",
+    "disclaimer": "Market data is provided by one or more third party data sources, including CoinGecko. Such third-party content is provided solely for informational purposes and should not be treated as advice to buy, sell, or use any particular asset. MetaMask does not suggest the use of this content for any particular purpose and is not responsible for its accuracy."
```
The first `third party` should be `third-party`

sethkfman commented on app/components/UI/AssetOverview/AboutAsset/AboutAsset.tsx:
```diff
@@ -0,0 +1,93 @@
+import { ThemeColors } from '@metamask/design-tokens/dist/js/themes/types';
+import i18n, { strings } from '../../../../../locales/i18n';
+
+const createStyles = (colors: ThemeColors) =>
```
Move styles out to a separate file

nikoferro committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into swaps/token-details

nikoferro committed:
chore: moved styles to external files, and a typo fix

nikoferro commented:
hey @sethkfman, following your feedback, i moved each of the new components styles to a `[ComponentName].styles.tsx` file. Also fixed the typo on the locale json

sethkfman commented on app/components/UI/AssetOverview/AboutAsset/AboutAsset.tsx:
```diff
@@ -0,0 +1,73 @@
+        <View>
+          <SkeletonPlaceholder>
+            <SkeletonPlaceholder.Item
+              width={'100%'}
```
Can you make this a common style that can be applied across all 3

nikoferro replied:
updating this

sethkfman commented on app/components/UI/AssetOverview/AssetOverview.tsx:
```diff
@@ -0,0 +1,312 @@
+import Price from './Price';
+import { SEND_BUTTON_ID } from '../../../../wdio/screen-objects/testIDs/Screens/WalletView.testIds';
+
+const createStyles = (colors: ThemeColors) =>
```
These styles should be broken out of the component

nikoferro replied:
updating this as part of @Cal-L feedback

sethkfman commented on app/components/UI/AssetOverview/Price/Price.tsx:
```diff
@@ -0,0 +1,139 @@
+  currentCurrency: string;
+  comparePrice: number;
+  isLoading: boolean;
+  timePeriod: '1d' | '7d' | '1w' | '1m' | '3m' | '1y' | '3y';
```
There is a type `TimePeriod` defined in AssetOverview.tsx I would put it in a common place and use it across these components

nikoferro replied:
updating this

sethkfman commented on app/components/UI/AssetOverview/PriceChart/PriceChart.tsx:
```diff
@@ -0,0 +1,254 @@
+      <Defs key="gradient">
+        <LinearGradient id="gradient" x1="0" y1="1" x2="0" y2="0">
+          <Stop
+            offset="0"
```
Create a single source for these repeated styles

nikoferro replied:
not sure i follow what u mean with this one since there are no repeated styles on the selected code. can you clarify this one?

sethkfman commented on app/components/Views/Asset/index.js:
```diff
@@ -384,6 +430,36 @@ class Asset extends PureComponent {
+    };
+
+    const goToSwaps = () => {
+      navigation.navigate('Swaps', {
```
Use the Routes here

nikoferro replied:
added a new value on the Routes file for this

Cal-L commented on app/components/hooks/useTokenHistoricalPrices.ts:
```diff
@@ -0,0 +1,52 @@
+    };
+    fetchPrices();
+  }, [address, chainId, timePeriod, from, to, vsCurrency]);
+  return { prices, isLoading, error };
```
Seems like the pattern from `useTokenDescriptions` is { data, ..., ... }. Seems cleaner to keep it consistent and name this data as well. Consumers can then reassign the name when they're using it

nikoferro replied:
updating this

Cal-L commented on app/components/UI/AssetOverview/PriceChart/utils.ts:
```diff
@@ -0,0 +1,41 @@
+}
+
+// this data draws a placeholder chart to show a greyed out chart when there is no data available
+export const placeholderData = [
```
Is this still needed if we're rendering a no data overlay over the chart?

nikoferro replied:
Yes because it renders a this fake chart

![Screenshot 2023-05-04 at 13 24 01](https://user-images.githubusercontent.com/8658960/236190571-fb13aec1-4f21-4bb8-9a7f-d38f81bbaf31.png)

Cal-L reviewed (commented):
Left some comments

nikoferro committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into swaps/token-details

nikoferro commented on app/util/number/index.js:
```diff
@@ -449,10 +450,39 @@ export function weiToFiat(
+          decimalString.match(/[1-9]/)[0],
+        );
+        if (firstNonZeroDecimal > 0) {
+          amount = parseFloat(amount).toFixed(firstNonZeroDecimal + 3);
```
Its not quite like that but you raise a good point. For 0.0000201, we actually show 0.0000201, which is expected. For 0.0000200 you are right that we are showing trailing zeros, so im removing those

0.0000201 -> 0.0000201
0.0000200 -> 0.00002

Cal-L replied:
Ah ok. What about 0.00002009? That becomes 0.0000200?

nikoferro commented on app/images/transaction-icons/swap-failed.png:
In this case im just extending the already available PNG icons for transactions. These are a set that are PNG in the current version. We can change them all to SVG but i would not include does changes on this PR.

nikoferro commented on app/components/UI/TransactionElement/index.js:
```diff
@@ -284,6 +314,8 @@ class TransactionElement extends PureComponent {
       tx: { time, status },
     } = this.props;
+    const colors = this.context.colors || mockTheme.colors;
+    const styles = createStyles(colors);
```
Cannot do this because this is a class component, but took the same approach done in the AccountApproval component which also expose the theme colors and typography

nikoferro commented on app/components/UI/TransactionElement/index.js:
```diff
@@ -77,18 +75,45 @@ const createStyles = (colors) =>
+      paddingTop: 0,
+    },
+    listItemTitle: {
+      fontSize: 16,
```
noted, updating this

nikoferro commented on app/components/Views/AssetOptions/index.tsx:
```diff
@@ -33,19 +36,27 @@ const createStyles = (colors: any) =>
-      justifyContent: 'center',
+      alignItems: 'flex-start',
+      justifyContent: 'flex-start',
+      display: 'flex',
```
updating this

nikoferro commented on app/components/Views/AssetOptions/index.tsx:
```diff
@@ -108,20 +119,22 @@ const AssetOptions = (props: Props) => {
           return (
             <View key={label}>
-              <View style={styles.divider} />
               <TouchableOpacity style={styles.optionButton} onPress={onPress}>
```
hey @Cal-L not really sure which icon you refer here. If you mean in general. This is how this component looks like
![Simulator Screen Shot - iPhone 11 - 2023-05-03 at 18 57 00](https://user-images.githubusercontent.com/8658960/235986676-bd7d4139-0ae2-45a8-9756-1e485f6de298.png)

Cal-L replied:
Yep, looking for a screenshot like this. Thanks!

nikoferro commented on app/components/Views/AssetOptions/index.tsx:
```diff
@@ -33,19 +36,27 @@ const createStyles = (colors: any) =>
+      justifyContent: 'flex-start',
+      display: 'flex',
+      flexDirection: 'row',
+      paddingHorizontal: 38,
```
This is the design i have for this, as you can see its 38px, but yeah we should def consolidate this 

![Screenshot 2023-05-03 at 18 59 31](https://user-images.githubusercontent.com/8658960/235987325-fb7336e6-96f2-499c-b85a-326a22761cd7.png)

nikoferro replied:
Updating this to match the new design you shared

Cal-L replied:
Thanks, let's also remove commented style properties

nikoferro replied:
oops, good catch

nikoferro commented on app/components/UI/TransactionElement/index.js:
```diff
@@ -294,21 +326,29 @@ class TransactionElement extends PureComponent {
+                    {fiatValue}
+                  </ListItem.FiatAmount>
                 )}
+                <ListItem.Amount style={styles.listItemAmount}>
```
hey @Cal-L the only change for this is swapping the order, i think this might be a pre existing bug
![Screenshot 2023-05-03 at 19 05 40](https://user-images.githubusercontent.com/8658960/235988762-fdd03339-db8a-4ee5-bdb0-8188c6ebd993.png)

The changes for this section are just cosmetic, and only affect the font sizing and swapping the position of the token value and the fiat value.
![Simulator Screen Shot - iPhone 11 - 2023-05-03 at 19 09 33](https://user-images.githubusercontent.com/8658960/235989908-aeea9536-67bd-4125-aa7e-100e9f2f733b.png)

nikoferro commented on app/components/UI/Transactions/index.js:
```diff
@@ -459,6 +465,13 @@ class Transactions extends PureComponent {
+    const { nativeEvent } = event;
+    const { contentOffset } = nativeEvent;
+    // 16 is the top padding of the list
+    this.props.onScrollThroughContent(contentOffset.y);
```
good catch, added a check if that handler exists

nikoferro commented on app/components/UI/Transactions/index.js:
```diff
@@ -74,15 +74,19 @@ const createStyles = (colors) =>
+    },
+    disclaimerText: {
+      color: colors.text.alternative,
+      fontSize: 12,
```
updating this

nikoferro commented on app/components/UI/Transactions/index.js:
```diff
@@ -616,6 +629,25 @@ class Transactions extends PureComponent {
 
+  renderDisclaimer = () => {
+    const colors = this.context.colors || mockTheme.colors;
+    const styles = createStyles(colors);
```
Cannot do this because this is a class component, but took the same approach done in the AccountApproval component which also expose the theme colors and typography

nikoferro commented on app/components/Views/Asset/ActivityHeader.styles.tsx:
```diff
@@ -0,0 +1,20 @@
+    },
+    title: {
+      fontSize: 16,
+      lineHeight: 24,
```
updating this

nikoferro commented on app/components/Views/Asset/ActivityHeader.tsx:
```diff
@@ -0,0 +1,23 @@
+import { Asset } from '../../UI/AssetOverview/AssetOverview.types';
+import createStyles from './ActivityHeader.styles';
+
+const styles = createStyles();
```
updating this

nikoferro commented on app/components/UI/AssetOverview/AssetActionButton/AssetActionButton.tsx:
```diff
@@ -0,0 +1,105 @@
+  testID,
+}: AssetActionButtonProps) => {
+  const { colors } = useTheme();
+  const styles = createStyles(colors);
```
updating this

nikoferro commented on app/components/UI/AssetOverview/AssetActionButton/AssetActionButton.tsx:
```diff
@@ -0,0 +1,105 @@
+    switch (type) {
+      case 'send': {
+        return (
+          <MaterialCommunityIcon
```
I did this change initially and was told on an initial feature review w/design to undo these changes. Would def approach this on a separate PR

Cal-L replied:
May I ask who asked to undo and why - designers should already be using the icon component available in the Figma library.

nikoferro commented on app/components/UI/AssetOverview/AssetOverview.tsx:
```diff
@@ -0,0 +1,312 @@
+    warning: {
+      borderRadius: 8,
+      color: colors.text.default,
+      ...fontStyles.normal,
```
updating this

nikoferro commented on app/components/UI/AssetOverview/AssetOverview.tsx:
```diff
@@ -0,0 +1,312 @@
+  });
+
+  const { colors = mockTheme.colors } = useContext(ThemeContext);
+  const styles = useMemo(() => createStyles(colors), [colors]);
```
updating this

nikoferro commented on app/components/UI/AssetOverview/AssetOverview.tsx:
```diff
@@ -0,0 +1,312 @@
+            timePeriod={timePeriod}
+          />
+          <View style={styles.chartNavigationWrapper}>
+            {(['1d', '1w', '1m', '3m', '1y', '3y'] as TimePeriod[]).map(
```
updating this

nikoferro commented on app/components/UI/AssetOverview/AssetOverview.tsx:
```diff
@@ -0,0 +1,312 @@
+                  label={strings(
+                    `asset_overview.chart_time_period_navigation.${label}`,
+                  )}
+                  onPress={handleSelectTimePeriod.bind(this, label)}
```
updating this

nikoferro commented on app/components/UI/AssetOverview/AboutAsset/AboutAsset.tsx:
```diff
@@ -0,0 +1,73 @@
+
+  const { data: descriptions, isLoading: isDescriptionLoading } =
+    useTokenDescriptions({
+      address: asset.address || zeroAddress(),
```
This follows the navigation logic since the address is present in the params except in the case of zero addresses. Going to change this to `asset.isETH ? zeroAddress() : asset.address` to have the isETH safeguard but the same underlying logic that comes from the routing applies here

Cal-L replied:
Does `useTokenDescriptions` return data when zero address is provided or does it just return null and the about section wouldn't render?

nikoferro replied:
Resolving this one since this is a review on top of code that changed. Current code is this one 
``` 
const { data: descriptions, isLoading: isDescriptionLoading } =
    useTokenDescriptions({
      address: asset.isETH ? zeroAddress() : asset.address,
      chainId: chainId as string,
    });
```

If zero address is provided it returns data related to the native currrency
In any case this section is hidden for now since we need to curate the data

nikoferro commented on app/components/UI/AssetOverview/AboutAsset/AboutAsset.styles.tsx:
```diff
@@ -0,0 +1,24 @@
+    title: {
+      fontSize: 16,
+      lineHeight: 24,
+      fontWeight: 'bold',
```
updating this

nikoferro commented on app/components/UI/AssetOverview/AboutAsset/ContentDisplay.styles.tsx:
```diff
@@ -0,0 +1,21 @@
+    },
+    disclaimer: {
+      color: colors.text.alternative,
+      fontSize: 12,
```
updating this

nikoferro commented on app/components/UI/AssetOverview/AboutAsset/ContentDisplay.tsx:
```diff
@@ -0,0 +1,47 @@
+      {disclaimer && isExpanded && (
+        <Text style={styles.disclaimer}>{disclaimer}</Text>
+      )}
+      <TouchableOpacity onPress={toggleContent}>
```
updating this

nikoferro commented on app/components/UI/AssetOverview/AboutAsset/ContentDisplay.tsx:
```diff
@@ -0,0 +1,47 @@
+  disclaimer,
+}: ContentDisplayProps) => {
+  const { colors = mockTheme.colors } = useContext(ThemeContext);
+  const styles = useMemo(() => createStyles(colors), [colors]);
```
updating this

nikoferro commented on app/components/UI/AssetOverview/Balance/Balance.tsx:
```diff
@@ -0,0 +1,29 @@
+}
+
+const Balance = ({ balance, fiatBalance }: BalanceProps) => {
+  const { colors = mockTheme.colors } = useContext(ThemeContext);
```
updating this

nikoferro commented on app/components/UI/AssetOverview/ChartNavigationButton/ChartNavigationButton.tsx:
```diff
@@ -0,0 +1,30 @@
+  selected,
+}: ChartNavigationButtonProps) => {
+  const { colors = mockTheme.colors } = useContext(ThemeContext);
+  const styles = useMemo(
```
updating

nikoferro commented on app/components/UI/AssetOverview/Price/Price.styles.tsx:
```diff
@@ -0,0 +1,51 @@
+      paddingHorizontal: 16,
+    },
+    symbol: {
+      fontSize: 12,
```
updating this

nikoferro commented on app/components/UI/AssetOverview/Price/Price.tsx:
```diff
@@ -0,0 +1,139 @@
+  isLoading,
+  timePeriod,
+}: PriceProps) => {
+  const { colors = mockTheme.colors } = useContext(ThemeContext);
```
updating this

nikoferro commented on app/components/UI/AssetOverview/Price/Price.tsx:
```diff
@@ -0,0 +1,139 @@
+              <Icon
+                name={
+                  diff > 0
+                    ? 'trending-up'
```
Tried to, but in this particular case is not possible since the TrendUp TrendDown icons we have on our library are squared and do not match the design. Maybe we can create new variants for these on our library, but would tackle this on a separate PR

Cal-L replied:
Fine for now. For future reference - if Figma designs do not use the Icon component, let's ask our designers to update their designs.

nikoferro commented on app/components/UI/AssetOverview/PriceChart/PriceChart.styles.tsx:
```diff
@@ -0,0 +1,46 @@
+      paddingTop: 10,
+    },
+    noDataOverlay: {
+      position: 'absolute',
```
updating this

nikoferro commented on app/components/UI/AssetOverview/PriceChart/PriceChart.styles.tsx:
```diff
@@ -0,0 +1,46 @@
+      padding: 96,
+      zIndex: 1,
+    },
+    noDataOverlayTitle: {
```
updating this

nikoferro commented on app/components/UI/AssetOverview/PriceChart/PriceChart.tsx:
```diff
@@ -0,0 +1,254 @@
+      </Text>
+      <Title style={styles.noDataOverlayTitle}>No chart data</Title>
+      <Text style={styles.noDataOverlayText}>
+        We could not fetch any data for this token
```
good catch, fixed this. forgot to add these two to the json file

nikoferro commented on app/components/UI/AssetOverview/PriceChart/PriceChart.tsx:
```diff
@@ -0,0 +1,254 @@
+
+  return (
+    <View style={styles.chart}>
+      <View style={styles.chartArea} {...panResponder.current.panHandlers}>
```
solved using a similar approach, was not able to make the waitFor related solution to work as expected so i tried something suggested by `react-native-gesture-handler` also

Cal-L replied:
TODO in future - let's convert this code to use both react-native-gesture-handler and react-native-reanimated instead. It'll remove the need to manage states and function calls for gestures.

nikoferro commented on app/components/UI/AssetOverview/AboutAsset/ContentDisplay.tsx:
```diff
@@ -0,0 +1,47 @@
+      )}
+      <TouchableOpacity onPress={toggleContent}>
+        <Text style={styles.button}>
+          {isExpanded ? 'Show less' : 'Show more'}
```
updating this

nikoferro commented on app/components/UI/AssetOverview/PriceChart/PriceChart.tsx:
```diff
@@ -0,0 +1,254 @@
+        <AreaChart
+          style={styles.chartArea}
+          data={chartHasData ? priceList : placeholderData}
+          // TODO: figure out why test fails when using curveMonotoneX
```
updating this

nikoferro commented on app/components/UI/AssetOverview/Price/Price.tsx:
```diff
@@ -0,0 +1,139 @@
+  };
+
+  const timePeriodTextDict: Record<
+    '1d' | '7d' | '1w' | '1m' | '3m' | '1y' | '3y',
```
updating this

nikoferro committed:
chore: updates related to code review

nikoferro committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into swaps/token-details

nikoferro committed:
fix: update to test case

nikoferro committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into swaps/token-details

wachunei commented on app/components/hooks/useTokenHistoricalPrices.ts:
```diff
@@ -0,0 +1,54 @@
+    const fetchPrices = async () => {
+      setIsLoading(true);
+      try {
+        const baseUri = 'https://price-api.metafi.codefi.network/v1';
+        const fromTo = from && to ? `&from=${from}&to=${to}` : '';
+        const uri = `${baseUri}/chains/${chainId}/historical-prices/${address}?timePeriod=${
+          timePeriod === '1w' ? '7d' : timePeriod
+        }&vsCurrency=${vsCurrency}${fromTo}`;
```
use `URL` object and `searchParams.set` to set params for readability:
```ts
const uri = new URL(`${baseUri}/chains/${chainId}/historical-prices/${address});
uri.searchParams.set('timePeriod', timePeriod === '1w' ? '7d': timePeriod);
uri.searchParams.set('vsCurrency', vsCurrency);
if (from && to) {
  uri.searchParams.set('from', from);
  uri.searchParams.set('to', to);
}
 ```

nikoferro replied:
updating this and `useTokenDescriptions`

nikoferro committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into swaps/token-details

nikoferro commented on app/components/UI/Navbar/index.js:
```diff
@@ -1081,7 +1090,10 @@ export function getNetworkNavbarOptions(
           // eslint-disable-next-line no-mixed-spaces-and-tabs
         )
       : () => <View />,
-    headerStyle: innerStyles.headerStyle,
```
sorry for the delay on this one because i had to ask design. but i am updating this and replacing the back for the X

Cal-L replied:
Since this util is used in other places, could. you verify that it still renders properly for the rest of the screens? I think we'll need a conditional here

nikoferro committed:
chore: last parts of PR feedback

Cal-L commented on app/component-library/components/Icons/Icon/assets/document-code.svg:
Ah. Instead of scaling down the group size, could we ask our designer to scale up the inside instead? The DS team expects the default group sizes to be 24.

nikoferro replied:
got the new svg, updating this

Cal-L commented on app/component-library/components/Icons/Icon/assets/received.svg:
Same question above

Cal-L commented on app/components/Views/Asset/index.js:
```diff
@@ -149,10 +191,15 @@ class Asset extends PureComponent {
   };
 
+  onScrollThroughContent = (contentOffset = 0) => {
+    this.updateNavBar(contentOffset);
```
TODO - aim to use react-native-reanimated in the future

nikoferro committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into swaps/token-details

nikoferro committed:
feat: second round of pr feedback

nikoferro committed:
fix: better footer btns alignment

nikoferro committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into swaps/token-details

nikoferro committed:
fix: fix for header shadows on empty tx list

nikoferro committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into swaps/token-details

nikoferro committed:
Merge branch 'main' into swaps/token-details

nikoferro committed:
Merge branch 'main' into swaps/token-details

nikoferro committed:
chore: needed lodash version

nikoferro committed:
chore: snapshot updated

nikoferro committed:
Merge branch 'main' into swaps/token-details

wachunei committed:
chore: resolve merge conflicts

wachunei committed:
Merge branch 'main' into swaps/token-details

wachunei committed:
test: update snapshot

wachunei commented:
@SocketSecurity ignore-all

sethkfman reviewed (approved):
LGTM
